File: OrchidCore/src/main/java/com/eden/orchid/impl/server/admin/lists/GeneratorsList.java
package com.eden.orchid.impl.server.admin.lists;

import com.eden.orchid.api.generators.OrchidGenerator;
import com.eden.orchid.api.server.admin.AdminList;

import javax.inject.Inject;
import java.util.Collection;
import java.util.Set;
import java.util.TreeSet;

public final class GeneratorsList implements AdminList<OrchidGenerator> {

    private Set<OrchidGenerator> list;

    @Inject
    public GeneratorsList(Set<OrchidGenerator> list) {
        this.list = new TreeSet<>(list);
    }

    @Override
    public String getKey() {
        return "generators";
    }

    @Override
    public Collection<OrchidGenerator> getItems() {
        return list;
    }

    @Override
    public String getItemId(OrchidGenerator item) {
        return item.getClass().getSimpleName();
    }

    @Override
    public OrchidGenerator getItem(String id) {
        for(OrchidGenerator item : getItems()) {
            if(id.equals(item.getClass().getSimpleName())) {
                return item;
            }
        }
        return null;
    }
}

File: plugins/OrchidPluginDocs/src/main/kotlin/com/eden/orchid/pluginDocs/lists/GeneratorsList.kt
package com.eden.orchid.pluginDocs.lists

import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.server.admin.AdminList
import java.util.TreeSet
import javax.inject.Inject

@JvmSuppressWildcards
class GeneratorsList @Inject
constructor(list: Set<OrchidGenerator>) : AdminList {

    private val list: Set<OrchidGenerator>

    init {
        this.list = TreeSet(list)
    }

    override fun getKey(): String {
        return "generators"
    }

    override fun getItems(): Collection<Any> {
        return list
    }

    override fun getItemId(item: Any): String {
        return item.javaClass.simpleName
    }

    override fun getItem(id: String): Any? {
        for (item in items) {
            if (id == item.javaClass.simpleName) {
                return item
            }
        }
        return null
    }
}

